univention: add common code for univention corporate server modules

keachi opened the pull request:
##### ISSUE TYPE
- Feature Pull Request
##### ANSIBLE VERSION

```
ansible 2.2.0
```
##### SUMMARY

This is common code for univention corporate server modules:
actually there are some PR on the ansible-modules-extras repository:
- [udm_dns_zone](https://github.com/ansible/ansible-modules-extras/pull/2382)
- [udm_dns_record](https://github.com/ansible/ansible-modules-extras/pull/2394)
- [udm_group](https://github.com/ansible/ansible-modules-extras/pull/2396)
- [udm_user](https://github.com/ansible/ansible-modules-extras/pull/2406)
- [udm_share](https://github.com/ansible/ansible-modules-extras/pull/2439)

Additional modules will follow with reference to this PR.

Commit:
univention: add common code for univention corporate server modules

bcoca commented on lib/ansible/module_utils/univention.py:
```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
```
this file is an import, no shebang needed

bcoca commented on lib/ansible/module_utils/univention.py:
```diff
@@ -0,0 +1,214 @@
+    import univention.admin.uldap
+    import univention.admin.objects
+    import univention.admin.config
+    import re
```
not all these imports belong here, they should have specific messages if dependencies other than univention are missing

Commit:
univention: try import only univention specific libraries

Commit:
Code Review with @2-B, slight API changes and refactoring.

* Added module documentation overview, describing the provided functions
* Moved module-global objects into getter functions, so that we don't
  need to import possibly-unavailable univention modules at the module level.
* Renamed some exports for improved consistency:

  - module_name()    -> module_by_name()
  - orig_ldap        -> ldap_module()
  - ldap             -> uldap()

Note that this introduces slight API changes from the outside. Instead of
directly accessing module properties, you now have module functions with the
same name. Examples:

   - ansible.module_utils.univention.position_base_dn()
   - ansible.module_utils.univention.config_registry()
   - ansible.module_utils.univention.base_dn()
   - ansible.module_utils.univention.config()

Commit:
module_utils univention: fix library

Commit:
move module_utils from univention to univention_umc, because python import univention fails if library is called univention

2-B commented:
We changed the whole import process, because the import should be executed on the target machine.
I removed the shebang.

keachi commented:
I removed the shebang, thanks for reviewing.
The whole import should be executed on the target machine, so we changed the import process.

Commit:
univention_umc: fix intention

resmo commented:
You used GPL, which is fine for module but for module_utils, a BSD style license is preferred . The reason is that closed source modules can be distributed using the code in module_utils.

keachi commented:
I/we prefer GPL, because we want, that other modules for univention are also open source.

resmo commented:
I understand, I am ok with it but would like to get the ok from ansible core. /cc @bcoca

resmo commented:
I brought this up to the Public Core Meeting (see log https://meetbot.fedoraproject.org/ansible-meeting/2016-08-09/ansible_core_public_meeting.2016-08-09-19.07.html)

I am afraid, Ansible insists on all items below module_utils/ not being GPL.

resmo commented:
As an option you could keep GPL but move your shared code to `ansible-modules-extras/univention/__init__.py` and adjust your include to it in your modules e.g. `from ansible.modules.extras.univention import ...`

resmo commented:
ignore my last comment, this won't work. Never mind.

Commit:
univention: change common code to BSD-2-clause

keachi commented:
I got the rights to relicense to BSD-2-clause.

resmo commented:
kicking the build again

resmo commented:
LGTM
